src/js_of_ocaml_rules.ml

```diff
@@ -4,14 +4,19 @@ open Build.O
 
 module SC = Super_context
 
-let dev_mode sctx = SC.profile sctx = "dev"
+let jsoo_compilation = SC.Env.jsoo_compilation
 
-let separate_compilation_enabled = dev_mode
+let pretty sctx ~dir =
+  match jsoo_compilation sctx ~dir with
+  | Separate ->  ["--pretty"]
```

Comment:
`--pretty` and `--sourcemap` have no relationship with separate compilation. There should be enabled with the dev profile.

Reply:
Yup that is true. I just kept the ild behavior to keep things simple for an initial PR. As I said in my ticket, the plan is to treat these as normal default flags for dev profile